lib/Sema/TypeCheckConcurrency.h

```diff
@@ -246,6 +246,17 @@ struct ActorReferenceResult {
   operator Kind() const { return kind; }
 };
 
+/// Specifies whether checks applied to function types should
+/// apply to their params, results, or both
+enum class FunctionCheckType {
```

Comment:
i like the use of an enum class here. my suggestion is to rename it to FunctionCheckKind to avoid any confusion with this representing a Type in the typechecker.